rosette/api.py

```diff
@@ -455,24 +455,26 @@ def call(self, parameters):
         @param parameters: An object specifying the data,
         and possible metadata, to be processed by the endpoint.  See the
         details for those object types.
-        @type parameters: For C{name-translation}, L{NameTranslationParameters},
-        otherwise L{DocumentParameters} or L{str}
+        @type parameters: Parameters types or L{str} for document request.
+        @param paramtype: Required parameters type.
         @return: A python dictionary expressing the result of the invocation.
         """
+        if paramtype and not isinstance(parameters, paramtype):
+            raise RosetteException(
+                "incompatible",
+                "The parameters must be " + str(paramtype),
+                self)
 
-        if not isinstance(parameters, _DocumentParamSetBase):
-            if self.suburl != self.api.endpoints['NAME_SIMILARITY'] \
-                    and self.suburl != self.api.self.api.endpoints['NAME_TRANSLATION'] \
-                    and self.suburl != self.api.self.api.endpoints['NAME_DEDUPLICATION'] \
-                    and self.suburl != self.api.self.api.endpoints['ADDRESS_SIMILARITY']:
-                text = parameters
-                parameters = DocumentParameters()
-                parameters['content'] = text
-            else:
-                raise RosetteException(
-                    "incompatible",
-                    "Text-only input only works for DocumentParameter endpoints",
-                    self.suburl)
+        if type(parameters) == str:
+            text = parameters
+            parameters = DocumentParameters()
+            parameters['content'] = text
+
+        if not isinstance(parameters, _RequestParametersBase):
```

Comment:
This lets names parameter types through, which then fail with badrequesformat. Example:

`api = rs.API(service_url='http://localhost:8181/rest/v1')
    params = rs.NameTranslationParameters()
    params["name"] = "some data to translate"
    params["targetLanguage"] = "eng"
    print(api.sentences(params))`

We could do something like `if not paramtype and not isinstance(parameters, DocumentParameters):`, which would still let names parameters through for names endpoints but fail them on document endpoints